src/modules/networks/useNetworks.ts

```diff
@@ -1,54 +1,90 @@
 import { useCallback, useEffect } from 'react';
-import { useQuery } from '@tanstack/react-query';
+import { useQuery, hashQueryKey } from '@tanstack/react-query';
 import { queryClient } from 'src/ui/shared/requests/queryClient';
 import { emitter } from 'src/ui/shared/events';
-import { networksStore } from 'src/modules/networks/networks-store.client';
+import {
+  getNetworksStore,
+  mainNetworksStore,
+  testenvNetworksStore,
+} from 'src/modules/networks/networks-store.client';
+import { usePreferences } from 'src/ui/features/preferences';
+import { invariant } from 'src/shared/invariant';
 import { getNetworksBySearch } from '../ethereum/chains/requests';
 import type { ChainId } from '../ethereum/transactions/ChainId';
+import { NetworksStore } from './networks-store';
 
-export function useNetworks(chainIds?: string[]) {
+function useNetworksStore() {
+  const { preferences } = usePreferences();
```

Comment:
Not sure how critical this is, but looks like this hook can just return boolean that shows which store should be used. And in the useNetworks we will use needed store. This can optimise dependency compare logic